lib/elixir/lib/task/supervised.ex

```diff
@@ -211,26 +215,43 @@ defmodule Task.Supervised do
           case waiting do
             %{^position => {_, {:ok, _} = ok}} -> Map.put(waiting, position, {nil, ok})
             %{^position => {_, :running}} -> Map.put(waiting, position, {nil, {:exit, reason}})
+            %{^position => {_, :timed_out}} -> Map.put(waiting, position, {nil, {:exit, :timeout}})
```

Comment:
I thought about that, I wasn't sure. I don't have any argument for `:timed_out` here though, so I updated to use `:timeout`. Should I change the message we send with `Process.send_after/3` as well?